integrationtests/subscribe_update_test.go

```diff
@@ -0,0 +1,489 @@
+		startGroup  uint64
+		startObject uint64
+		endGroup    uint64
+	}{
+		{"zero_values", 0, 0, 0},
+		{"start_only", 100, 50, 0},
+		{"end_only", 0, 0, 200},
+		{"both", 100, 50, 200},
+		{"large_values", 1<<40, 1<<30, 1<<50},
+	}
+	
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			msg := &moqtransport.Message{
+				Method:             moqtransport.MessageSubscribeUpdate,
+				RequestID:          1,
+				SubscriberPriority: 128,
+				StartGroup:         tc.startGroup,
+				StartObject:        tc.startObject,
+				EndGroup:           tc.endGroup,
+				Forward:            1,
+			}
+			
+			assert.Equal(t, tc.startGroup, msg.StartGroup)
+			assert.Equal(t, tc.startObject, msg.StartObject)
+			assert.Equal(t, tc.endGroup, msg.EndGroup)
+		})
+	}
+}
+
+// TestSubscribeUpdateEndGroupScenarios tests various EndGroup scenarios
+func TestSubscribeUpdateEndGroupScenarios(t *testing.T) {
+	testCases := []struct {
+		name        string
+		startGroup  uint64
+		startObject uint64
+		endGroup    uint64
+		description string
+	}{
+		{
+			name:        "unbounded_subscription",
+			startGroup:  100,
+			startObject: 0,
+			endGroup:    0,
+			description: "EndGroup 0 means no end limit",
+		},
+		{
+			name:        "bounded_subscription",
+			startGroup:  100,
+			startObject: 0,
+			endGroup:    500,
+			description: "Subscribe to groups 100-500",
+		},
+		{
+			name:        "single_group",
+			startGroup:  100,
+			startObject: 0,
+			endGroup:    100,
+			description: "Subscribe to single group only",
+		},
+		{
+			name:        "live_edge_catchup",
+			startGroup:  1000,
+			startObject: 0,
+			endGroup:    1050,
+			description: "Catch up with recent groups near live edge",
+		},
+		{
+			name:        "historical_range",
+			startGroup:  0,
+			startObject: 0,
+			endGroup:    100,
+			description: "Subscribe to historical data only",
+		},
+		{
+			name:        "max_uint64_end",
+			startGroup:  0,
+			startObject: 0,
+			endGroup:    ^uint64(0),
+			description: "Maximum possible end value",
+		},
+	}
+	
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			msg := &moqtransport.Message{
+				Method:             moqtransport.MessageSubscribeUpdate,
+				RequestID:          1,
+				SubscriberPriority: 128,
+				StartGroup:         tc.startGroup,
+				StartObject:        tc.startObject,
+				EndGroup:           tc.endGroup,
+				Forward:            1,
+			}
+			
+			// Verify the values are stored correctly
+			assert.Equal(t, tc.startGroup, msg.StartGroup, tc.description)
+			assert.Equal(t, tc.startObject, msg.StartObject, tc.description)
+			assert.Equal(t, tc.endGroup, msg.EndGroup, tc.description)
+			
+			// Additional validation could be added here based on business rules
+			// For example, checking that end >= start when end != 0
+		})
+	}
+}
+
+// TestSubscribeUpdateNarrowingPattern tests the pattern of narrowing subscriptions
+func TestSubscribeUpdateNarrowingPattern(t *testing.T) {
```

Comment:
This looks a bit more tricky but I also don't think it makes sense. It just seems to verify the values set in some testcases are narrower than in others but it doesn't actually test any code?